File: educational-system-consumer-admin-9000/src/main/java/edu/hubu/consumeradmin/service/IAdminService.java
package edu.hubu.consumeradmin.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.admin.Admin;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "admin-provider-service")
public interface IAdminService {

    @GetMapping("/admin/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/admin/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "adminId") String orderBy);

    @GetMapping("/admin/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "adminId") String orderBy
    );

    @PostMapping("/admin/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "adminId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.Admin admin
    );

    @PostMapping("/admin/")
    CommonResult save(@RequestBody Admin admin);

    @PutMapping("/admin/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody Admin admin);

    @DeleteMapping("/admin/{id}")
    CommonResult logicDeleteById(@PathVariable("id") String id);

    @PatchMapping("/admin/{id}")
    CommonResult forbidById(@PathVariable("id") String id);

    @DeleteMapping("/admin/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/admin/IAdminService.kt
package edu.hubu.commons.service.admin

import edu.hubu.commons.model.admin.Admin;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface IAdminService {
    fun save(t: Admin): Admin
    fun logicalDeleteById(id: String): Boolean
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: Admin): Admin
    fun forbidById(id: String): Boolean
    fun selectOneById(id: String): Admin
    fun selectAllOrderBy(order: String = "adminId"): List<Admin>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "adminId"): List<Admin>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "adminId", t: edu.hubu.commons.mybatis.model.Admin): List<edu.hubu.commons.mybatis.model.Admin>
}
